File: recipes_3e/apache/paragraphs/paragraphs.py
#!/usr/bin/python
# paragraphs.py: generate HTML paragraphs

import cgi
import cookbook

title = "Query Output Display - Paragraphs"

# Print content type header, blank line that separates
# headers from page body, and first part of page

print("Content-Type: text/html")
print("")
print("<html>")
print("<head><title>%s</title></head>" % title)
print("<body>")

conn = cookbook.connect()

#@ _DISPLAY_PARAGRAPH_
cursor = conn.cursor()
cursor.execute("SELECT NOW(), VERSION(), DATABASE()")
(now, version, db) = cursor.fetchone()
cursor.close()
if db is None:
  db = 'NONE'
para = "Local time on the MySQL server is %s." % now
print("<p>%s</p>" % cgi.escape(para, 1))
para = "The server version is %s." % version
print("<p>%s</p>" % cgi.escape(para, 1))
para = "The default database is %s." % db
print("<p>%s</p>" % cgi.escape(para, 1))
#@ _DISPLAY_PARAGRAPH_

conn.close()

print("</body>")
print("</html>")

File: recipes_3e/apache/paragraphs/paragraphs.php
<?php
# paragraphs.php: generate HTML paragraphs

require_once "Cookbook.php";
require_once "Cookbook_Webutils.php";

$title = "Query Output Display - Paragraphs";
?>

<html>
<head><title><?php print ($title); ?></title></head>
<body>

<?php
$dbh = Cookbook::connect ();

#@ _DISPLAY_PARAGRAPH_1_
$sth = $dbh->query ("SELECT NOW(), VERSION(), DATABASE()");
list ($now, $version, $db) = $sth->fetch (PDO::FETCH_NUM);
if ($db === NULL)
  $db = "NONE";
$para = "Local time on the MySQL server is $now.";
print ("<p>" . htmlspecialchars ($para) . "</p>");
$para = "The server version is $version.";
print ("<p>" . htmlspecialchars ($para) . "</p>");
$para = "The default database is $db.";
print ("<p>" . htmlspecialchars ($para) . "</p>");
#@ _DISPLAY_PARAGRAPH_1_

$dbh = NULL;

# Display the paragraph with a lot of switching between modes
?>

<!-- _DISPLAY_PARAGRAPH_2_ -->
<p>Local time on the MySQL server is
<?php print (htmlspecialchars ($now)); ?>.</p>
<p>The server version is
<?php print (htmlspecialchars ($version)); ?>.</p>
<p>The default database is
<?php print (htmlspecialchars ($db)); ?>.</p>
<!-- _DISPLAY_PARAGRAPH_2_ -->

</body>
</html>
